pkg/install/vault/init.go

```diff
@@ -0,0 +1,212 @@
+	plugins := []*vaultapi.RegisterPluginInput{
+		{
+			Name:    "oauthapp",
+			Type:    consts.PluginTypeSecrets,
+			SHA256:  "2455ad9450e415efb42fd28c1436f4bf7e377524be5e534e55e6658b8ef56bd2",
+			Command: "vault-plugin-secrets-oauthapp-v3.0.0-beta.3-linux-amd64",
+		},
+	}
+
+	secretEngines := []*model.VaultSecretEngine{
+		{
+			Name: vi.vaultCoreConfig.LogServicePath,
+			Type: model.VaultSecretEngineTypeKVV2,
+		},
+		{
+			Name: vi.vaultCoreConfig.TenantPath,
+			Type: model.VaultSecretEngineTypeKVV2,
+		},
+		{
+			Name: vi.vaultCoreConfig.TransitPath,
+			Type: model.VaultSecretEngineTypeTransit,
+		},
+	}
+
+	policyGen := &vaultPolicyGenerator{
+		LogServicePath: vi.vaultCoreConfig.LogServicePath,
+		TenantPath:     vi.vaultCoreConfig.TenantPath,
+		TransitKey:     vi.vaultCoreConfig.TransitKey,
+		TransitPath:    vi.vaultCoreConfig.TransitPath,
+	}
+
+	policies := []*model.VaultPolicy{
+		{
+			Name:  vi.vaultCoreConfig.LogServiceVaultAgentRole,
+			Rules: string(policyGen.logServicePolicy()),
+		},
+		{
+			Name:  vi.vaultCoreConfig.MetadataAPIVaultAgentRole,
+			Rules: string(policyGen.metadataAPIPolicy()),
+		},
+		{
+			Name:  vi.vaultCoreConfig.OperatorVaultAgentRole,
+			Rules: string(policyGen.operatorPolicy()),
+		},
+	}
+
+	auth, err := vi.vaultSystemManager.GetAuthMethod(fmt.Sprintf("%s/", vi.vaultConfig.JWTMount))
+	if err != nil {
+		return err
+	}
+
+	if auth != nil {
+		policyGen.AuthJWTAccessor = auth.Accessor
+
+		// FIXME Add installer configuration for `metadata-api-tenant`
+		policies = append(policies, &model.VaultPolicy{
+			Name:  "metadata-api-tenant",
+			Rules: string(policyGen.metadataAPITenantPolicy()),
+		})
+	}
+
+	// FIXME Do not hardcode the service account names
+	kubernetesRoles := []*model.VaultKubernetesRole{
+		{
+			Name:                          vi.vaultCoreConfig.LogServiceVaultAgentRole,
+			BoundServiceAccountNames:      []string{"relay-core-v1-log-service-vault-agent"},
+			BoundServiceAccountNamespaces: []string{vi.vaultConfig.Namespace},
+			Policies:                      []string{vi.vaultCoreConfig.LogServiceVaultAgentRole},
+			TTL:                           "24h",
+		},
+		{
+			Name:                          vi.vaultCoreConfig.MetadataAPIVaultAgentRole,
+			BoundServiceAccountNames:      []string{"relay-core-v1-metadata-api-vault-agent"},
+			BoundServiceAccountNamespaces: []string{vi.vaultConfig.Namespace},
+			Policies:                      []string{vi.vaultCoreConfig.MetadataAPIVaultAgentRole},
+			TTL:                           "24h",
+		},
+		{
+			Name:                          vi.vaultCoreConfig.OperatorVaultAgentRole,
+			BoundServiceAccountNames:      []string{"relay-core-v1-operator-vault-agent"},
+			BoundServiceAccountNamespaces: []string{vi.vaultConfig.Namespace},
+			Policies:                      []string{vi.vaultCoreConfig.OperatorVaultAgentRole},
+			TTL:                           "24h",
+		},
+	}
+
+	// FIXME Add installer configuration for `metadata-api-tenant`
```

Comment:
Does this still need to be done? I don't think it will work correctly without it.

Reply:
The comment is indicating that we need to add an installer configuration for the name instead of hardcoding it. We have similar options for other key names, but not this one.

So it does need to be done, but it does work correctly as of now. It's been fully tested (_ad nauseam_ at this point). The naming we use for staging/production is slightly different currently (i.e. `nebula/system-metadata-api-tenant`) so we need to add this recommended configuration option or adjust accordingly.

This is fine to leave for an additional improvement.

Reply:
Ahh gotcha. Didn't realize it was just referring to customizing the name.